File: Trie/1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int)-> List[int]:
        seenSum = {}

        for i, num in enumerate(nums):
            if num in seenSum:
                return([i, seenSum[num]])
            
            seenSum[target-num] =i

        return []

             

File: Trie/1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let seen = {}
    let out = [-1,-1]
    for(let i=0;i<nums.length;i++) {
        num = nums[i]
        if(target-num in seen) {
            out = [i, seen[target-num ]]
            break;
        }
        seen[num] = i
    }
   

    return out
};